src/tactics/FStar.Tactics.V2.Basic.fst

```diff
@@ -2120,12 +2139,51 @@ let refl_typing_builtin_wrapper (f:unit -> 'a) : tac (option 'a & issues) =
       }) in
       [issue], None
   in
+
+  (* Deep compress the guards since we are about to roll back the UF.
+  The caller will discharge them if needed. *)
+  let gs =
+    if Some? r then
+      List.map (fun (e,g) -> e, SC.deep_compress false g) (snd (Some?.v r))
```

Comment:
A very minor nit, for readability we could write it as: `let allow_uvars = false in SC.deep_compress allow_uvars ...` etc.